lib/Interpreter/Value.cpp

```diff
@@ -231,9 +231,15 @@ namespace cling {
   namespace valuePrinterInternal {
     std::string printTypeInternal(const Value& V);
     std::string printValueInternal(const Value& V);
+    extern const char* const kUndefined;
   } // end namespace valuePrinterInternal
 
   void Value::print(llvm::raw_ostream& Out, bool Escape) const {
+    if (!m_Interpreter) {
+      Out << valuePrinterInternal::kUndefined << "\n";
```

Comment:
I would prefer if the different "failure" cases had different output strings, instead of all mapping to `kUndefined`, e.g. here: `Out << "<ERROR: missing Interpreter!>\n"`.

This will also remove the awkwardness of having to redeclare `kUndefined` in different, seemingly unrelated places - runtime and compiled code, Value.cpp and ValuePrinter.cpp.